packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js

```diff
@@ -850,6 +852,84 @@ describe('#unit BoltProtocolV4x3', () => {
         expect(unpackedDateTimeWithoutOffset).toEqual(object)
       })
 
+      it('should pack and unpack DateTimeWithOffset', () => {
+        fc.assert(
+          fc.property(
+            fc.date({
+              min: temporalUtil.newDate(utils.MIN_UTC_IN_MS + utils.ONE_DAY_IN_MS),
+              max: temporalUtil.newDate(utils.MAX_UTC_IN_MS - utils.ONE_DAY_IN_MS)
+            }),
+            utils.arbitraryTimeZoneId(),
```

Comment:
Consider adding another `fc.integer({min: 0, max: 999_999})` and add it on top of `date.getUTCMilliseconds() * 1_000_000`. This way, you cover the full ns range.